fhir-path/src/main/java/com/ibm/fhir/path/function/registry/FHIRPathFunctionRegistry.java

```diff
@@ -0,0 +1,170 @@
+import com.ibm.fhir.path.function.ConvertsToIntegerFunction;
+import com.ibm.fhir.path.function.ConvertsToQuantityFunction;
+import com.ibm.fhir.path.function.ConvertsToStringFunction;
+import com.ibm.fhir.path.function.CountFunction;
+import com.ibm.fhir.path.function.DescendantsFunction;
+import com.ibm.fhir.path.function.DistinctFunction;
+import com.ibm.fhir.path.function.EmptyFunction;
+import com.ibm.fhir.path.function.EndsWithFunction;
+import com.ibm.fhir.path.function.ExcludeFunction;
+import com.ibm.fhir.path.function.ExtensionFunction;
+import com.ibm.fhir.path.function.FHIRPathFunction;
+import com.ibm.fhir.path.function.FirstFunction;
+import com.ibm.fhir.path.function.GetValueFunction;
+import com.ibm.fhir.path.function.HasValueFunction;
+import com.ibm.fhir.path.function.HtmlChecksFunction;
+import com.ibm.fhir.path.function.IndexOfFunction;
+import com.ibm.fhir.path.function.IntersectFunction;
+import com.ibm.fhir.path.function.IsDistinctFunction;
+import com.ibm.fhir.path.function.ItemFunction;
+import com.ibm.fhir.path.function.LastFunction;
+import com.ibm.fhir.path.function.LengthFunction;
+import com.ibm.fhir.path.function.LowerFunction;
+import com.ibm.fhir.path.function.MatchesFunction;
+import com.ibm.fhir.path.function.MemberOfFunction;
+import com.ibm.fhir.path.function.NotFunction;
+import com.ibm.fhir.path.function.NowFunction;
+import com.ibm.fhir.path.function.ReplaceFunction;
+import com.ibm.fhir.path.function.ReplaceMatchesFunction;
+import com.ibm.fhir.path.function.ResolveFunction;
+import com.ibm.fhir.path.function.SingleFunction;
+import com.ibm.fhir.path.function.SkipFunction;
+import com.ibm.fhir.path.function.SliceFunction;
+import com.ibm.fhir.path.function.StartsWithFunction;
+import com.ibm.fhir.path.function.SubsetOfFunction;
+import com.ibm.fhir.path.function.SubstringFunction;
+import com.ibm.fhir.path.function.SubsumedByFunction;
+import com.ibm.fhir.path.function.SubsumesFunction;
+import com.ibm.fhir.path.function.SupersetOfFunction;
+import com.ibm.fhir.path.function.TailFunction;
+import com.ibm.fhir.path.function.TakeFunction;
+import com.ibm.fhir.path.function.ToBooleanFunction;
+import com.ibm.fhir.path.function.ToCharsFunction;
+import com.ibm.fhir.path.function.ToDateFunction;
+import com.ibm.fhir.path.function.ToDateTimeFunction;
+import com.ibm.fhir.path.function.ToDecimalFunction;
+import com.ibm.fhir.path.function.ToIntegerFunction;
+import com.ibm.fhir.path.function.ToQuantityFunction;
+import com.ibm.fhir.path.function.ToStringFunction;
+import com.ibm.fhir.path.function.TodayFunction;
+import com.ibm.fhir.path.function.TypeFunction;
+import com.ibm.fhir.path.function.UnionFunction;
+import com.ibm.fhir.path.function.UpperFunction;
+
+public class FHIRPathFunctionRegistry {
+    private static final FHIRPathFunctionRegistry INSTANCE = new FHIRPathFunctionRegistry();
+    private Map<String, FHIRPathFunction> functionMap = new ConcurrentHashMap<>();
```

Comment:
small thing... do you intend this to mutable? e.g. a downstream consumer could change this.

Reply:
discussed with John - no direct access

Reply:
Added `final` keyword to FHIRPathFunctionRegistry class declaration.